multicluster/controllers/multicluster/resourceexport_controller.go

```diff
@@ -47,16 +72,305 @@ type ResourceExportReconciler struct {
+	return client.IgnoreNotFound(err)
+}
+
+func (r *ResourceExportReconciler) updateEndpointResourceImport(ctx context.Context,
+	existRe *mcsv1alpha1.ResourceExport, resImpName types.NamespacedName) error {
+	resImport := &mcsv1alpha1.ResourceImport{}
+	err := r.Client.Get(ctx, resImpName, resImport)
+	if err != nil {
+		klog.ErrorS(err, "fail to get ResourceImport", "resourceimport", resImpName)
+		return client.IgnoreNotFound(err)
+	}
+	newResImport, changed, err := r.refreshEndpointsResourceImport(existRe, resImport, false)
+	if err != nil {
+		return err
+	}
+	if changed {
+		if err = r.handleUpdateEvent(ctx, resImpName, newResImport, existRe); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (r *ResourceExportReconciler) getExistResImport(ctx context.Context,
+	resExport mcsv1alpha1.ResourceExport, resImportName types.NamespacedName) (bool, *mcsv1alpha1.ResourceImport, error) {
+	importedResNameSpace := resExport.Labels["sourceNamespace"]
+	importedResName := resExport.Labels["sourceName"]
+	var createResImport bool
+	existResImport := &mcsv1alpha1.ResourceImport{}
+	err := r.Client.Get(ctx, resImportName, existResImport)
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			klog.ErrorS(err, "fail to get ResourceImport", "resourceimport", resImportName.String())
+			return createResImport, nil, err
+		}
+		existResImport = &mcsv1alpha1.ResourceImport{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      resImportName.Name,
+				Namespace: resImportName.Namespace,
+			},
+			Spec: mcsv1alpha1.ResourceImportSpec{
+				ClusterIDs: []string{},
+				Name:       importedResName,
+				Namespace:  importedResNameSpace,
+			},
+		}
+		createResImport = true
+	}
+	return createResImport, existResImport, nil
+}
+
+func (r *ResourceExportReconciler) refreshServiceResourceImport(
+	resExport *mcsv1alpha1.ResourceExport,
+	resImport *mcsv1alpha1.ResourceImport,
+	createResImport bool) (*mcsv1alpha1.ResourceImport, bool, error) {
+	newResImport := resImport.DeepCopy()
+	newResImport.Spec.Name = resExport.Spec.Name
+	newResImport.Spec.Namespace = resExport.Spec.Namespace
+	newResImport.Spec.Kind = common.ServiceImportKind
+	if createResImport {
+		newResImport.Spec.ServiceImport = &mcs.ServiceImport{
+			Spec: mcs.ServiceImportSpec{
+				Ports: svcPortsConverter(resExport.Spec.Service.ServiceSpec.Ports),
+				Type:  mcs.ClusterSetIP,
+			},
+		}
+		return newResImport, true, nil
+	}
+	// TODO: check ClusterIPs difference if it is being used in ResrouceImport later
+	convertedPorts := svcPortsConverter(resExport.Spec.Service.ServiceSpec.Ports)
+	if !apiequality.Semantic.DeepEqual(newResImport.Spec.ServiceImport.Spec.Ports, convertedPorts) {
+		undeletedItems, err := r.getUndeletedResourceExportItems(resExport)
+		if err != nil {
+			klog.ErrorS(err, "fail to list ResourceExports, retry later")
+			return newResImport, false, err
+		}
+		// when there is only one Service ResourceExport, ResourceImport should reflect the change
```

Comment:
they didn't handle this case, MCSAPI repo only create a sample for Service creation.

Reply:
What you think we return an error and let controller retry (until conflict is resolved)?

Reply:
I think this kind conflict can only be handled manually, do you mean this kind of situation should be corrected by admin when they do service export, so we can let controller retry after admin resolve conflict?

Reply:
done, I leave the logic to update ResourceImport when there is only one ResourceExport and for other cases, it will return error. let me know if it's suitable or not.